test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -2236,6 +2236,146 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
             $pathologicalRatio | Should -BeGreaterThan 5
         }
     }
+
+    Context 'Invoke-WebSession: Connection persistence in a WebSession' {
+        $connWarning = [regex]::new('WebSession.+HTTP')
+
+        function RunCheckingPersistence {
+            param(
+                [uri]$Uri,
+                [string]$Command,
+                [object]$Session,
+                [switch]$LostPersistence,
+                [switch]$CaptureSession
+            )
+
+            $pwsh = [PowerShell]::Create()
+            $pwsh.Runspace.SessionStateProxy.SetVariable('uri', $Uri)
+            if ($Session) {
+                $pwsh.Runspace.SessionStateProxy.SetVariable('Session', $Session)
+                $command = "$command -WebSession `$Session"
+            }
+            if ($CaptureSession) {
+                $command = "$command -SessionVariable Session"
+            }
+            $script = "`$null = $command -Verbose"
+            $pwsh.AddScript($script).Invoke()
+            $session = $pwsh.Runspace.SessionStateProxy.GetVariable('Session')
+
+            $expectedWarnings = if ($LostPersistence) { 1 } else { 0 }
+            ($pwsh.Streams.Verbose | Where-Object { $connWarning.Matches($_.Message) }).Count | Should -Be $expectedWarnings
```

Comment:
Please update the variable name - it is not for `warning`.

Reply:
Changed variables to reflect the message better:
* `$expectedConnRecreatedCount` for the 0 or 1 count of messages
* `$matchConnRecreatedMessage` for the regex
* `-ExpectConnectionRecreated` for the function parameter